### PR TITLE
Add more Raspberry Pi Pico LCD Shields by PiMoroni

### LINKED ISSUE #239: [HW] PiMoroni Raspberry Pi Pico LCD Modules as Shields
hardware-gh&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;hardware-gh

**Is this request related to a missing driver support for a particular hardware platform, SoC or board? Please describe.**
I need support for PiMoroni Pico LCD Modules as shields. The following shield should be supported:

- [x] PiMoroni [PIM543 Pico Display Pack](https://pimoroni.com/picodisplay), 1.44" 65K colors, 135 × 240 pixels, driver: [ST7789](https://docs.zephyrproject.org/latest/build/dts/api/bindings/display/sitronix%2Cst7789v.html#dtbinding-sitronix-st7789v)
      https://github.com/pimoroni/pimoroni-pico/blob/main/libraries/pico_display/pico_display.hpp
- [x] PiMoroni [PIM580 Pico Display Pack 2.0](https://pimoroni.com/picodisplay20), 2" 65K colors, 240 × 320 pixels, driver: [ST7789](https://docs.zephyrproject.org/latest/build/dts/api/bindings/display/sitronix%2Cst7789v.html#dtbinding-sitronix-st7789v)
      https://github.com/pimoroni/pimoroni-pico/blob/main/libraries/pico_display_2/pico_display_2.hpp

In this early first stage the Zephyr upstream samples "Shell on USB console", "LCD Orientation and Bit Order Test" and "LVGL Basic Sample" should be usable.

**Describe why you are asking for this support?**
With this support the board+shield variation for display based systems will scaling up.

I'm preparing a changeset for PR. I'll maintain the new board for Bridle and later contribute and maintain to/in Zephyr upstream.

**Additional context**
The new shields support will coming together with RPi Pico R3 connector support as currently provided by Bridle only.

### PR DESCRIPTION
**Pull request by rexut**
Fixes #239 and also brings many improvements in shield documentation:

- Pico LCD 1.44 shield by PiMoroni [PIM543 Pico Display Pack](https://pimoroni.com/picodisplay)
- Pico LCD 2 shield by PiMoroni [PIM580 Pico Display Pack 2.0](https://pimoroni.com/picodisplay20)

### CODE REVIEW

File: boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_1_44.overlay [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+			       INPUT_KEY_1>;
+		lvgl-codes = <LV_KEY_LEFT LV_KEY_RIGHT
+			      LV_KEY_ENTER>;
+	};
+
+	gpio_keys: gpio_keys {
+		compatible = "gpio-keys";
+		button_0: button_0 {
+			label = "User button A";
+			zephyr,code = <INPUT_KEY_0>;
```
Comment (rexut) [inline boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_1_44.overlay:45]: **praise (blocking):** change to `INPUT_KEY_A`

File: boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_1_44.overlay [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+		};
+		button_1: button_1 {
+			label = "User button B";
+			zephyr,code = <INPUT_KEY_1>;
```
Comment (rexut) [inline boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_1_44.overlay:49]: **praise (blocking):** change to `INPUT_KEY_B`

File: boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_1_44.overlay [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+		};
+		button_2: button_2 {
+			label = "User button X";
+			zephyr,code = <INPUT_KEY_2>;
```
Comment (rexut) [inline boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_1_44.overlay:53]: **praise (blocking):** change to `INPUT_KEY_X`

File: boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_1_44.overlay [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+		};
+		button_3: button_3 {
+			label = "User button Y";
+			zephyr,code = <INPUT_KEY_3>;
```
Comment (rexut) [inline boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_1_44.overlay:57]: **praise (blocking):** change to `INPUT_KEY_Y`

File: boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_2.overlay [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+			       INPUT_KEY_1>;
+		lvgl-codes = <LV_KEY_LEFT LV_KEY_RIGHT
+			      LV_KEY_ENTER>;
+	};
+
+	gpio_keys: gpio_keys {
+		compatible = "gpio-keys";
+		button_0: button_0 {
+			label = "User button A";
+			zephyr,code = <INPUT_KEY_0>;
```
Comment (rexut) [inline boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_2.overlay:45]: **praise (blocking):** change to `INPUT_KEY_A`

File: boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_2.overlay [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+		};
+		button_1: button_1 {
+			label = "User button B";
+			zephyr,code = <INPUT_KEY_1>;
```
Comment (rexut) [inline boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_2.overlay:49]: **praise (blocking):** change to `INPUT_KEY_B`

File: boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_2.overlay [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+		};
+		button_2: button_2 {
+			label = "User button X";
+			zephyr,code = <INPUT_KEY_2>;
```
Comment (rexut) [inline boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_2.overlay:53]: **praise (blocking):** change to `INPUT_KEY_X`

File: boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_2.overlay [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+		};
+		button_3: button_3 {
+			label = "User button Y";
+			zephyr,code = <INPUT_KEY_3>;
```
Comment (rexut) [inline boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_2.overlay:57]: **praise (blocking):** change to `INPUT_KEY_Y`

File: boards/shields/rpi_pico_lcd/waveshare_pico_lcd_0_96.overlay [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+		lvgl-codes = <LV_KEY_LEFT LV_KEY_RIGHT
+			      LV_KEY_UP LV_KEY_DOWN
+			      LV_KEY_ENTER>;
+	};
+
+	gpio_keys: gpio_keys {
+		compatible = "gpio-keys";
+		button_0: button_0 {
+			label = "User button A";
+			zephyr,code = <INPUT_KEY_0>;
```
Comment (rexut) [inline boards/shields/rpi_pico_lcd/waveshare_pico_lcd_0_96.overlay:47]: **praise (blocking):** change to `INPUT_KEY_A`

File: boards/shields/rpi_pico_lcd/waveshare_pico_lcd_0_96.overlay [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+		};
+		button_1: button_1 {
+			label = "User button B";
+			zephyr,code = <INPUT_KEY_1>;
```
Comment (rexut) [inline boards/shields/rpi_pico_lcd/waveshare_pico_lcd_0_96.overlay:51]: **praise (blocking):** change to `INPUT_KEY_B`

File: boards/shields/rpi_pico_lcd/waveshare_pico_lcd_1_3.overlay [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+		lvgl-codes = <LV_KEY_LEFT LV_KEY_RIGHT
+			      LV_KEY_UP LV_KEY_DOWN
+			      LV_KEY_ENTER>;
+	};
+
+	gpio_keys: gpio_keys {
+		compatible = "gpio-keys";
+		button_0: button_0 {
+			label = "User button A";
+			zephyr,code = <INPUT_KEY_0>;
```
Comment (rexut) [inline boards/shields/rpi_pico_lcd/waveshare_pico_lcd_1_3.overlay:47]: **praise (blocking):** change to `INPUT_KEY_A`

File: boards/shields/rpi_pico_lcd/waveshare_pico_lcd_1_3.overlay [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+		};
+		button_1: button_1 {
+			label = "User button B";
+			zephyr,code = <INPUT_KEY_1>;
```
Comment (rexut) [inline boards/shields/rpi_pico_lcd/waveshare_pico_lcd_1_3.overlay:51]: **praise (blocking):** change to `INPUT_KEY_B`

File: boards/shields/rpi_pico_lcd/waveshare_pico_lcd_1_3.overlay [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+		};
+		button_2: button_2 {
+			label = "User button X";
+			zephyr,code = <INPUT_KEY_2>;
```
Comment (rexut) [inline boards/shields/rpi_pico_lcd/waveshare_pico_lcd_1_3.overlay:55]: **praise (blocking):** change to `INPUT_KEY_X`

File: boards/shields/rpi_pico_lcd/waveshare_pico_lcd_1_3.overlay [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+		};
+		button_3: button_3 {
+			label = "User button Y";
+			zephyr,code = <INPUT_KEY_3>;
```
Comment (rexut) [inline boards/shields/rpi_pico_lcd/waveshare_pico_lcd_1_3.overlay:59]: **praise (blocking):** change to `INPUT_KEY_Y`

**Comment by rexut** [comment]
see also review comments in PR #238

File: boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_1_44.overlay (hunk 1 above)
Comment (rexut) [reply boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_1_44.overlay:45]: done

File: boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_1_44.overlay (hunk 3 above)
Comment (rexut) [reply boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_1_44.overlay:53]: done

File: boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_2.overlay (hunk 5 above)
Comment (rexut) [reply boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_2.overlay:45]: done

File: boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_1_44.overlay (hunk 2 above)
Comment (rexut) [reply boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_1_44.overlay:49]: done

File: boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_1_44.overlay (hunk 4 above)
Comment (rexut) [reply boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_1_44.overlay:57]: done

File: boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_2.overlay (hunk 6 above)
Comment (rexut) [reply boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_2.overlay:49]: done

File: boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_2.overlay (hunk 7 above)
Comment (rexut) [reply boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_2.overlay:53]: done

File: boards/shields/rpi_pico_lcd/waveshare_pico_lcd_0_96.overlay (hunk 9 above)
Comment (rexut) [reply boards/shields/rpi_pico_lcd/waveshare_pico_lcd_0_96.overlay:47]: done

File: boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_2.overlay (hunk 8 above)
Comment (rexut) [reply boards/shields/rpi_pico_lcd/pimoroni_pico_lcd_2.overlay:57]: done

File: boards/shields/rpi_pico_lcd/waveshare_pico_lcd_0_96.overlay (hunk 10 above)
Comment (rexut) [reply boards/shields/rpi_pico_lcd/waveshare_pico_lcd_0_96.overlay:51]: done

File: boards/shields/rpi_pico_lcd/waveshare_pico_lcd_1_3.overlay (hunk 11 above)
Comment (rexut) [reply boards/shields/rpi_pico_lcd/waveshare_pico_lcd_1_3.overlay:47]: done

File: boards/shields/rpi_pico_lcd/waveshare_pico_lcd_1_3.overlay (hunk 12 above)
Comment (rexut) [reply boards/shields/rpi_pico_lcd/waveshare_pico_lcd_1_3.overlay:51]: done

File: boards/shields/rpi_pico_lcd/waveshare_pico_lcd_1_3.overlay (hunk 13 above)
Comment (rexut) [reply boards/shields/rpi_pico_lcd/waveshare_pico_lcd_1_3.overlay:55]: done

File: boards/shields/rpi_pico_lcd/waveshare_pico_lcd_1_3.overlay (hunk 14 above)
Comment (rexut) [reply boards/shields/rpi_pico_lcd/waveshare_pico_lcd_1_3.overlay:59]: done

**Comment by rexut** [comment]
> see also review comments in PR #238

done

**Comment by rexut** [comment]
> You can find the documentation preview for this PR at this [link](https://bridle.tiac-systems.net/ghpr/doc/PR-243 ). It will be updated about 10 minutes after the documentation build succeeds.
> 
> _Note: This comment is automatically posted by the Documentation Publishing GitHub Action._

Found the new shields here: https://bridle.tiac-systems.net/ghpr/doc/PR-243/bridle/boards/shields/rpi_pico_lcd/doc/index.html

**Review by tobiaskaestner — DISMISSED** [review]
LGTM

**Review by tobiaskaestner — APPROVED** [review]
Lgtm

**Pull request merged**